.github/workflows/AI.yml

```diff
@@ -0,0 +1,82 @@
+name: PR Summary and Code Check
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  summarize_and_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip
```

Comment:
The command `python -m pip install --upgrade pip` should be followed by a command to install the required packages in a single line to avoid potential issues with package installation order.